boring/Cargo.toml

```diff
@@ -19,24 +19,27 @@ rustdoc-args = ["--cfg", "docsrs"]
 [features]
 # Controlling the build
 
-# Use a FIPS-validated version of BoringSSL. This feature sets "fips-compat".
-fips = ["fips-compat", "boring-sys/fips"]
-
-# Use a FIPS build of BoringSSL, but don't set "fips-compat".
+# NOTE: This feature is deprecated. It panders to the submoduled
+# boringssl-fips, which is extremely old and requires modifications to the
+# bindings, as some newer APIs don't exist and some function signatures have
+# changed. It is highly recommended to use `fips-precompiled` instead.
 #
-# As of boringSSL commit a430310d6563c0734ddafca7731570dfb683dc19, we no longer
-# need to make exceptions for the types of BufLen, ProtosLen, and ValueLen,
-# which means the "fips-compat" feature is no longer needed.
+# This feature sets `fips-compat` on behalf of the user to guarantee bindings
+# compatibility with the submoduled boringssl-fips.
 #
-# TODO(cjpatton) Delete this feature and modify "fips" so that it doesn't imply
-# "fips-compat".
-fips-no-compat = ["boring-sys/fips"]
+# Use a FIPS-validated version of BoringSSL.
+fips = ["fips-compat", "boring-sys/fips"]
```

Comment:
Question: do we plan to modify/delete the `fips` feature as part of boring 5? I would be in favor of replacing `fips` with `fips-precompiled`, since this will allow internal users to migrate to the latest fips build without changing their feature set.

On the other hand, users of old fips builds will have to change their feature set, but there are less such users.

Reply:
Yeah ideally in 5.0 `fips` will become `fips-precompiled`